### PR TITLE
armv7m/imxrt: add new syspage implementation

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

This PR is a part of changes made in [HBuczynski/syspage](https://github.com/phoenix-rtos/phoenix-rtos-kernel/commits/HBuczynski/syspage) branch. In order to provide better readability of review, the changes were separated into smaller pieces.

The following changes consist of:
 - making common `hal.h` header for all targets
 - updating `pmap` in `armv7m/imxrt`
 - updating `map_init` & `page_init`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/armv7m/imxrt/pmap.c [hunk 1]
```diff
@@ -71,109 +56,20 @@ addr_t pmap_getMaxVAdrr(void)
...
+	return 0;
 }
 
 
 void _pmap_init(pmap_t *pmap, void **vstart, void **vend)
 {
-	(*vstart) = (void *)(((addr_t)(syspage->kernel.bss + syspage->kernel.bsssz + 1024 + 256) + 7) & ~7);
+	addr_t size = (addr_t)&_end - (addr_t)&__bss_start;
+
+	(*vstart) = (void *)(((addr_t)((addr_t)&__bss_start + size + 1024 + 256) + 7) & ~7);
```
Comment (agkaminski) [inline hal/armv7m/imxrt/pmap.c:67]: Hmm, we can fetch initial stack size from a first vector in the vector table.

File: vm/map.c [hunk 2]
```diff
@@ -1052,30 +1052,36 @@ void vm_mapGetStats(size_t *allocsz)
...
+		if (map == prog->dmaps[i])
+			return map_common.maps[i];
+	}
 #endif
+
 	return NULL;
 }
 
 
+#ifdef NOMMU
```
Comment (agkaminski) [inline vm/map.c:1073]: Move this `ifdef` to the inside of the function, make it return zero in `else` case

File: vm/map.c [hunk 3]
```diff
@@ -1193,12 +1194,12 @@ int _map_init(vm_map_t *kmap, vm_object_t *kernel, void **bss, void **top)
 
 	(*bss) += poolsz;
 	lib_printf("vm: Initializing memory mapper: (%d*%d) %d\n", map_common.nfree, sizeof(map_entry_t), poolsz);
-
-
+#ifdef NOMMU
```
Comment (agkaminski) [inline vm/map.c:1197]: We don't need this `ifdef` if `_map_mapsInit` returns zero for MMU targets

File: hal/armv7m/imxrt/pmap.c (hunk 1 above)
Comment (HBuczynski) [reply hal/armv7m/imxrt/pmap.c]: Done

File: vm/map.c (hunk 2 above)
Comment (HBuczynski) [reply vm/map.c]: Done

File: vm/map.c (hunk 3 above)
Comment (HBuczynski) [reply vm/map.c]: Done

**Review by agkaminski — APPROVED** [review]

**Pull request merged**